libbeat/cmd/instance/beat.go

```diff
@@ -886,3 +923,17 @@ func obfuscateConfigOpts() []ucfg.Option {
 		ucfg.ResolveNOOP,
 	}
 }
+
+func chooseMonitoringConfig(beatCfg beatConfig) (monitoringCfg *common.Config, err error) {
+	if beatCfg.MonitoringNew.Enabled() && beatCfg.Monitoring.Enabled() {
+		err = fmt.Errorf("both xpack.monitoring.* and monitoring.* cannot be set. Prefer to set monitoring.* and set monitoring.hosts to monitoring cluster hosts")
+	} else if beatCfg.Monitoring.Enabled() {
+		cfgwarn.Deprecate("7.0", "xpack.monitoring.* settings are deprecated. Use monitoring.* instead, but set monitoring.hosts to monitoring cluster hosts")
+		monitoringCfg = beatCfg.Monitoring
+		monitoringCfg.SetString("format", -1, "production")
+	} else {
+		monitoringCfg = beatCfg.MonitoringNew
+		monitoringCfg.SetString("format", -1, "monitoring")
+	}
+	return
```

Comment:
style:
- do not use fmt.Errorf with a constant string. Prefer a global variable initialized with `errors.New`.
- for longer if-else cascade you can use `switch { }`
- it's ok to use named return values for documentation purposes, but better avoid naked returns all together. ([This example](https://play.golang.org/p/6D3tIYLBz0T) displays the problem with naked returns and shadowing).